temporal/tests/test_e2e.py

```diff
@@ -1,14 +1,31 @@
 # (C) Datadog, Inc. 2023-present
 # All rights reserved
 # Licensed under a 3-clause BSD style license (see LICENSE)
-
 import pytest
 
 from datadog_checks.dev.utils import get_metadata_metrics
+from datadog_checks.temporal import TemporalCheck
 
+from .common import TAGS
+
+pytestmark = [pytest.mark.e2e]
+
+
+def test_e2e(dd_agent_check, instance):
+    aggregator = dd_agent_check(instance, rate=True)
 
-@pytest.mark.e2e
-def test_e2e(dd_agent_check):
-    aggregator = dd_agent_check({})
-    aggregator.assert_all_metrics_covered()
     aggregator.assert_metrics_using_metadata(get_metadata_metrics())
+
+    for metric in get_metadata_metrics():
+        aggregator.assert_metric(name=metric, at_least=0, tags=TAGS)
+    assert len(aggregator.metric_names) > 100
```

Comment:
Is `at_least=0` a little too loose of a requirement? I see we are also asserting `len(aggregator.metric_names) > 100` but I'm not too familiar with what metrics show up in the e2e env.

Reply:
The requirement that I'm trying to express here is that we get at least 101 metrics out of those listed in the metadata csv file (I think I was typically getting > 250). I'm using `aggregator.assert_metric` simply to populate `aggregator.metric_names`, that's the reason behind `at_least=0`. I could try to track down a small subset of metric names that are always there, but I feel like that would have more potential for flakiness and I'm not sure we'd gain much. 

There's really not much logic to test in the integration beyond verifying the mapping of metric names from Prometheus to datadog metrics, and we're already testing a subset of that explicitly in the "unit" tests. Thus, for e2e / integration testing, I think I'm happy with just "sanity checking" that there's nothing obviously wrong with the integration. But maybe there are other things I should consider?

Reply:
Are there known metrics in `metadata.csv` that may not show up in a regular setup of Temporal OpenMetrics endpoint? My concern with just checking the number of metrics > 100 is if a customer asks why they don't see `temporal.server.x` from their setup but it's listed in `metadata.csv`, and we can't confirm if that metric is being collected since we don't assert it in our own testing.

Reply:
Well, there are 600+ metrics listed in the metadata.csv file, of which the current testing environment produces less than 300. So with such large numbers, I don't see a point in being more specific.

If a customer doesn't see a specific metric, it's either because Temporal is not emitting it in their environment, or because the metric mapping is wrong. That's easy to find out by curling the metrics endpoint.